test/e2e/k8s_util.go

```diff
@@ -79,7 +82,9 @@ func (k *KubernetesUtils) GetPodsByLabel(ns string, key string, val string) ([]v
 	if err != nil {
 		return nil, errors.WithMessage(err, "unable to list pods")
 	}
+	k.podLock.Lock()
```

Comment:
I think it needs to protect access at L77 as well.

Reply:
done